src/main/java/seedu/module/ui/CommandBox.java

```diff
@@ -36,7 +36,7 @@ private void handleCommandEntered() {
         try {
             commandExecutor.execute(commandTextField.getText());
             commandTextField.setText("");
-        } catch (CommandException | ParseException e) {
+        } catch (CommandException | ParseException | RuntimeException e) {
```

Comment:
Maybe you can catch this exception inside the command, and the command can throw a commandException? Let's practice separation of concerns

Reply:
this exception is thrown from the constructor of the link. which is called in the parser. the link command class is not responsible for link construction.

Reply:
Ok, then maybe you can have the parser catch the exception and throw a ParseException then?